core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -834,24 +803,24 @@ oppia.directive('conversationSkin', [
                     }
                     FatigueDetectionService.reset();
                     NumberAttemptsService.reset();
-                    _nextFocusLabel = FocusManagerService.generateFocusLabel();
+                    _nextFocusLabel = focusLabel;
 
-                    PlayerTranscriptService.setDestination(newStateName);
+                    PlayerTranscriptService.setDestination(
+                      nextCard.getStateName());
 
                     // These are used to compute the dimensions for the
                     // next card.
-                    $scope.upcomingStateName = newStateName;
-                    $scope.upcomingParams = newParams;
-                    $scope.upcomingContentHtml = (
-                      contentHtml + _getRandomSuffix());
+                    $scope.nextCard = nextCard;
+                    $scope.nextCard.addRandomSuffixToContentHtml(
+                      _getRandomSuffix());
+                    $scope.nextCard.addRandomSuffixToInteractionHtml(
+                      _getRandomSuffix());
 
                     var _isNextInteractionInline = (
                       ExplorationPlayerStateService.isNextInteractionInline());
                     $scope.upcomingInlineInteractionHtml = (
                       _isNextInteractionInline ?
-                        ExplorationPlayerStateService.getNextInteractionHtml(
-                          _nextFocusLabel
-                        ) + _getRandomSuffix() : '');
+                        $scope.nextCard.getInteractionHtml() : '');
                     $scope.upcomingInteractionInstructions = (
                       ExplorationPlayerStateService.
                         getNextInteractionInstructions()
```

Comment:
why not get this from nextCard?

Reply:
Similar comment as to previous one, would require INTERACTION_SPECS in state card.

Reply:
OK to import.

Reply:
Done.